app/main.py

```diff
@@ -1,2 +1,111 @@
+from typing import Any
+
+
+class HashNode:
+    def __init__(
+            self, key: int | float | complex | str | tuple | object,
+            value: Any
+    ) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash(self.key)
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.loaded = int(self.capacity * (2 / 3))
+        self.hash_table = [None] * self.capacity
+
+    def get_hash_table_load_info(self) -> str:
+        return (
+            "is Load" if self.loaded == self.hash_table.count(None)
+            else "is not Load"
+        )
+
+    @staticmethod
+    def check_key_type(key: Any) -> None:
+        if isinstance(key, (set, dict, list)):
+            raise TypeError(f"unhashable type: '{type(key)}'")
+
+    def input_hash_data(self, node_pair: HashNode) -> None:
+        index = node_pair.hash % self.capacity
+        if self.hash_table[index] is None:
+            self.hash_table[index] = node_pair
+        elif self.hash_table[index].key == node_pair.key:
+            self.hash_table[index].value = node_pair.value
+        else:
+            while self.hash_table[index] is not None:
+                if self.hash_table[index].key == node_pair.key:
+                    self.hash_table[index].value = node_pair.value
+                    return
+                index = (index + 1) % self.capacity
+            self.hash_table[index] = node_pair
+
+    def resize_hash_table(self) -> None:
+        self.hash_table.extend([None] * self.capacity)
+        self.capacity *= 2
+        self.loaded = int(self.capacity * (2 / 3))
+        for i in range(self.capacity // 2):
+            if self.hash_table[i] is not None:
+                node_pair = self.hash_table[i]
+                self.hash_table[i] = None
+                self.input_hash_data(node_pair)
```

Comment:
In `resize_hash_table`, after extending the hash table, the existing nodes should be rehashed to their new positions. The current implementation may not correctly rehash nodes, leading to potential collisions or incorrect placements.